src/main/java/com/arpnetworking/metrics/impl/TsdMetricsFactory.java

```diff
@@ -107,37 +205,66 @@ protected TsdMetricsFactory(final Builder builder) {
     public static class Builder {
 
         /**
+         * Public constructor.
+         */
+        public Builder() {
```

Comment:
I've numbered your points in the ordered you listed so we can refer to them numerically in any further discussion.

1) Correct, you lose the named arguments benefit of the builder (e.g. 8 String args).
2) Of course. Any bug in the code should be caught as early as possible.
3) Please note that it is not considered a complete test unless you assert that results of whatever operation you performed. Exception-free execution on its own is an incomplete test case. Especially when you are fully aware that this library was designed and written to be exception neutral in its usage.
4) Yes it was discussed, and again for consistency we preferred not to throw an exception. For a equivalent example try to misconfigure Logback and let me know whether it kills your application.
5) This was considered as well. It can get out of hand when you have conditionally required members because it requires a sub-type for each required combination. I would be happy to approve this pattern if someone wants to step up and generate the builder type-tree at compile time by analyzing the annotations on the members. Otherwise, this is potentially a nightmare to create and maintain in all cases.